CMRF/Core/AbstractCall.php

```diff
@@ -208,7 +210,11 @@ protected function extractParameters($request) {
     }
 
     foreach (self::$protected as $field_name) {
-      if (isset($request[$field_name])) {
+      if ($field_name == 'action' && ($request['action'] ?? NULL) != ($this->getAction()) ) {
+        // Some actions may expect an 'action' key in the request parameters (e.g. "validate"),
+        // so don't filter that out if 'action' and '$request['action]' are different
+        continue;
```

Comment:
Hm, most actions that require an *action* parameter call that `api_action`, except for `validate`. So there's no case when the API action and the parameter `action` are the same?

Reply:
Can we guarantee that validate is the only Action that requires an action parameter? The only reason we didn't test for specifically validate is because there may be others we haven't thought of yet.

If we can guarantee it, more than happy to test specifically for validate, though!

Reply:
I haven't checked for other actions, I actually was just surprised that `validate` behaves differently here. I was trying to say that this condition assumes that no API action that has a parameter `action` can be called with both, the action and the `action` parameter, being the same.

And also, doesn't APIv3 treat the `action` parameter as the actual action?

I'm just looking at the code here, so maybe this is totally fine - just wanted to think out loud in case that rings a bell …

Reply:
> I actually was just surprised that validate behaves differently here

You and me both!

> And also, doesn't APIv3 treat the action parameter as the actual action?

Not as far as we experienced. CMRF here filters out the action parameter as a protected keyword. Do you suggest maybe it doesn't need to at all?